common/src/main/kotlin/org/imdc/extensions/common/DatasetExtensions.kt

```diff
@@ -73,6 +73,300 @@ object DatasetExtensions {
+            keywords,
+            arrayOf("dataset", "dataset2", "columnIndex", "columnIndex2"),
+            arrayOf(Dataset::class.java, Dataset::class.java, Int::class.java, Int::class.java),
+            "outerJoin",
+        )
+        val dataset = parsedArgs.requirePyObject("dataset").toJava<Dataset>()
+        val dataset2 = parsedArgs.requirePyObject("dataset2").toJava<Dataset>()
+        val column = parsedArgs.requirePyObject("columnIndex").toJava<Int>()
+        val column2 = parsedArgs.requirePyObject("columnIndex2").toJava<Int>()
+
+        val columnName = dataset.columnNames.toList()
+        val columnName2 = dataset2.columnNames.toList()
+        val combinedColumnName = columnName + columnName2
+
+        val columnType = dataset.columnTypes.toList()
+        val columnType2 = dataset2.columnTypes.toList()
+        val combinedColumnType = columnType + columnType2
+
+        val builder = DatasetBuilder.newBuilder()
+            .colNames(combinedColumnName)
+            .colTypes(combinedColumnType)
+
+        for (row in dataset.rowIndices) {
+            val listToAppend = arrayOfNulls<Any?>(combinedColumnName.size)
+            var row2: Int? = null
+
+            dataset2.rowIndices.forEachIndexed { rowIndex, _ ->
+                if (dataset[row, column] == dataset2[rowIndex, column2]) {
+                    row2 = rowIndex
+                    return@forEachIndexed
+                }
+            }
+
+            dataset.columnIndices.forEachIndexed { colIndex, _ ->
+                listToAppend[colIndex] = dataset[row, colIndex]
+            }
+
+            row2?.let { r2 ->
+                dataset2.columnIndices.forEachIndexed { colIndex, _ ->
+                    listToAppend[dataset.columnCount + colIndex] = dataset2[r2, colIndex]
+                }
+            }
+
+            builder.addRow(*listToAppend)
+        }
+
+        // Add unmatched rows from dataset2
+        for (row2 in dataset2.rowIndices) {
+            val listToAppend = arrayOfNulls<Any?>(combinedColumnName.size)
+            var row1: Int? = null
+
+            dataset.rowIndices.forEachIndexed { rowIndex, _ ->
+                if (dataset[rowIndex, column] == dataset2[row2, column2]) {
+                    row1 = rowIndex
+                    return@forEachIndexed
+                }
+            }
+
+            if (row1 == null) {
+                dataset2.columnIndices.forEachIndexed { colIndex, _ ->
+                    listToAppend[dataset.columnCount + colIndex] = dataset2[row2, colIndex]
+                }
+                builder.addRow(*listToAppend)
+            }
+        }
+
+        return builder.build()
+    }
+
+    @Suppress("unused")
+    @ScriptFunction(docBundlePrefix = "DatasetExtensions")
+    @KeywordArgs(
+        names = ["dataset", "columnsToSplit"],
+        types = [Dataset::class, Array<Array<Int>>::class],
+    )
+    fun splitter(args: Array<PyObject>, keywords: Array<String>): Array<Dataset?> {
```

Comment:
Returning as an array is a little weird - a plain list would be more typical for Python/Java/Ignition